cpp/emscripten_util.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <emscripten/bind.h>
+#include <string>
+
+std::string getExceptionMessage(intptr_t exceptionPtr) {
+  return std::string(reinterpret_cast<std::exception*>(exceptionPtr)->what());
+}
+
+EMSCRIPTEN_BINDINGS(Bindings) {
+  emscripten::function("getExceptionMessage", &getExceptionMessage);
+};
+
+#define TO_JSON(CLASSNAME,                                               \
+                INSTANCENAME) /* NOLINT(readability-identifier-naming)*/ \
+  void to_json(json& j, const CLASSNAME&(INSTANCENAME)) {                \
```

Comment:
What calls this? Is it inside of nlohmann::json?

Reply:
Yup. for example, this is why we in State, we can just do:
  void toJson(json& j) const { j = json{{"transform", _transform}}; }

That implicitly calls to_json on the point object to convert _transform to json.